### ISSUE TITLE
Does not work with debugtoolbar

### ISSUE THREAD

**Issue by jdimmerman** [alisaifee/flask-limiter#336]
The latest version does not work with flask_debugtoolbar due to https://github.com/flask-debugtoolbar/flask-debugtoolbar/issues/177.

In particular, in `manager.py`:
```
def blueprint_limits(self, app: flask.Flask, blueprint: str) -> List[Limit]:
        limits: List[Limit] = []

        blueprint_name = app.blueprints[blueprint].name if blueprint else None
```

`app.blueprints[blueprint]` will throw an `KeyError` if the blueprint name of the request doesn't exist in the set of blueprints for the app.

Of course debugtoolbar should fix their issue, but ideally this codebase would have protection. Something like (I would need to test this)
```
blueprint_name = app.blueprints[blueprint].name if blueprint and blueprint in app.blueprints else None
```

**Comment by alisaifee** [comment]
Thanks for the report and apologies for the regression. Looks like there's a few weird issues here:
1) [current_request.blueprint](https://github.com/pallets/flask/blob/a03719b01076a5bfdc2c8f4024eda7b874614bc1/src/flask/wrappers.py#L89) is pretty naive and simply splits the endpoint by `.` and returns the first part. I had not expected that so using `request.blueprint` as a strong indicator that the current request is indeed coming from a blueprint should be fixed (in flask-limiter).
2) The actual rule in debug toolbar that caused the error is actually a static route which technically should have been exempt from rate limits - but the exemption code naively only checks for `endpoint == "static"`. This has been there for ever - but I think it should probably be `endpoint.split(".")[-1] == "static"` 

I'll do a patch release for 2.3 since master has some other stuff which I want to do a minor version bump for.

**Comment by alisaifee** [comment]
This is resolved in [2.3.3](https://flask-limiter.readthedocs.io/en/stable/changelog.html#v2-3-3)

**Comment by jdimmerman** [comment]
Wow you are fast. Thanks @alisaifee!
